src/test/java/de/tum/cit/aet/artemis/shared/base/AbstractArtemisBuildAgentTest.java

```diff
@@ -0,0 +1,194 @@
+        "artemis.continuous-integration.image-cleanup.enabled=true", "artemis.continuous-integration.image-cleanup.disk-space-threshold-mb=1000000000",
+        "spring.liquibase.enabled=false", "artemis.version-control.ssh-private-key-folder-path=${java.io.tmpdir}", "artemis.version-control.build-agent-use-ssh=false",
+        "artemis.version-control.ssh-template-clone-url=ssh://git@locaFlhost:7921/", "artemis.continuous-integration.pause-grace-period-seconds=2" })
+public abstract class AbstractArtemisBuildAgentTest {
+
+    @Autowired
+    protected DockerClientTestService dockerClientTestService;
+
+    @Autowired
+    @Qualifier("hazelcastInstance")
+    protected HazelcastInstance hazelcastInstance;
+
+    @MockitoSpyBean
+    protected BuildAgentConfiguration buildAgentConfiguration;
+
+    @MockitoSpyBean
+    protected BuildJobGitService buildJobGitService;
+
+    protected DockerClient dockerClient;
+
+    private static DockerClient dockerClientMock;
+
+    private static final Path TEST_RESULTS_PATH = Path.of("src", "test", "resources", "test-data", "test-results");
+
+    private static final Path GRADLE_TEST_RESULTS_PATH = TEST_RESULTS_PATH.resolve("java-gradle");
+
+    protected static final Path PARTLY_SUCCESSFUL_TEST_RESULTS_PATH = GRADLE_TEST_RESULTS_PATH.resolve("partly-successful");
+
+    @BeforeAll
+    protected static void mockDockerClient() throws InterruptedException {
+        dockerClientMock = DockerClientTestService.mockDockerClient();
+    }
+
+    @BeforeEach
+    protected void mockServices() throws IOException {
+        mockBuildJobGitService();
+        dockerClientTestService.mockTestResults(dockerClientMock, PARTLY_SUCCESSFUL_TEST_RESULTS_PATH, LOCALCI_WORKING_DIRECTORY + LOCALCI_RESULTS_DIRECTORY);
+
+        // Mock the startContainerCmd to sleep for 100ms. This is necessary to appropriately test the build agent's behavior when a build job is started.
+        StartContainerCmd startContainerCmd = mock(StartContainerCmd.class);
+        when(dockerClientMock.startContainerCmd(anyString())).thenReturn(startContainerCmd);
+        doAnswer(invocation -> {
+            Thread.sleep(100);
+            return null;
+        }).when(startContainerCmd).exec();
+
+        when(buildAgentConfiguration.getDockerClient()).thenReturn(dockerClientMock);
+        dockerClient = dockerClientMock;
+    }
+
+    protected void mockBuildJobGitService() {
+        var objectId = mock(ObjectId.class);
+        doReturn("dummy-commit-hash").when(objectId).getName();
+        doReturn(objectId).when(buildJobGitService).getLastCommitHash(any());
+
+        var repository = mock(Repository.class);
+        doNothing().when(repository).closeBeforeDelete();
+        doReturn(Path.of("src", "test", "resources", "templates", "java", "java.txt")).when(repository).getLocalPath();
+        try {
+            doReturn(repository).when(buildJobGitService).cloneRepository(any(), any());
+        }
+        catch (Exception ignored) {
```

Comment:
is this something that should not happen? in case yes, i might add a log message if that would appear